server/src/main/java/org/opensearch/identity/noop/NoopIdentityPlugin.java

```diff
@@ -38,4 +46,9 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return new NoopTokenManager();
     }
+
+    @Override
+    public Subject getPluginSubject(Plugin plugin) {
```

Comment:
How would be distinguish "plugin subject" from "user subject" / "extension subject" / ... ?

Reply:
Based on the concrete implementation of Subject. For this PR, its only introducing the Subject for plugins. The value of Subject returned by this method will be passed into IdentityAwarePlugin's `assignSubject` and then the IdentityAwarePlugin can choose how to utilize it. I imagine a plugin dev would keep track of this subject through a naming convention like pluginSubject.

Reply:
Why not make it explicit with `PluginSubject` interface?

Reply:
Should there be a separate PluginSubject interface from the Subject interface? In the case of the PluginSubject, there isn't a notion of authenticate because its assumed if you are holding a PluginSubject that you can use it. Its secure in that plugins are given separate PluginSubjects to utilize.

I'll push a change to create a separate PluginSubject interface.